server/src/main/java/org/apache/druid/initialization/Initialization.java

```diff
@@ -518,7 +519,13 @@ private boolean shouldLoadOnCurrentNodeType(Object object)
       Set<NodeRole> rolesPredicate = Arrays.stream(loadScope.roles())
                                            .map(NodeRole::fromJsonName)
                                            .collect(Collectors.toSet());
-      return rolesPredicate.stream().anyMatch(nodeRoles::contains);
+      boolean shouldLoad = rolesPredicate.stream().anyMatch(nodeRoles::contains);
+      if (!shouldLoad) {
+        log.debug(
```

Comment:
Changed. As it turns out, it started at info and someone suggested it be moved to debug. I added it because, like you, I had a case in which an extension was not being loaded and needed to know why.